### ISSUE TITLE
Test execution freezes with a NodeJS version >= 10.12.0

### ISSUE THREAD

**Issue by Batta32** [yeoman/yeoman-test#52]
**Related issues**: #51, https://github.com/feathers-plus/generator-feathers-plus/issues/103, https://github.com/yeoman/generator/issues/1098 
**Environment**: Windows 10, NodeJS >= 10.12.0, mocha 5.2.0

When testing with the yeoman helper, I discovered that by making several hooks' calls to run the generator with prompts, it locks on the same point.

Example:
```javascript
beforeEach(async function() {
	await helpers.run(path.join(__dirname, '../generators/myGenerator'))
		.inDir(path.join(__dirname, 'tmp'))
		.withPrompts({
			name: 'aName',
			confirmation: true
	});
});
```
This happens using `CMD` and `PowerShell` (what is strange is that in each of them, the execution tests are locked in different points). However, using `bash`, the execution tests end successfully.

![image](https://user-images.githubusercontent.com/11904023/52488319-5016a800-2b9e-11e9-9e44-d681d977908f.png)

Debugging the tests, I found out that the `prompting` method runs fine and resolves the promise okay. 
The problem is that the `writing` method never runs at the second time that the `helpers.run` is called inside the `beforeEach` hook.

Finally, changing the prompts to arguments was the workaround that I found even though is not the final solution I want (this works in all the environments).

Example:
```javascript
beforeEach(async function() {
	await helpers.run(path.join(__dirname, '../generators/myGenerator'))
	.inDir(path.join(__dirname, 'tmp'))
	.withArguments(['-n', 'aName'])
});
```

![image](https://user-images.githubusercontent.com/11904023/52488394-7fc5b000-2b9e-11e9-8a11-a53ded931294.png)

**Comment by wictorwilen** [comment]
I have the same issue. All 10+ versions of node, on Windows, freezes as soon as I run any test similar to this:

```
    it('should generate bot project with v1.3 with unit tests', async () => {
        await helpers.run(testHelper.GENERATOR_PATH)
            .inDir(testHelper.TEMP_BOT_GENERATOR_PATH + '/bot01')
            .withArguments(['--no-telemetry'])
            .withPrompts({
                solutionName: 'bot-test-01',
                whichFolder: 'current',
                name: 'bottest01',
                developer: 'generator teams developer',
                manifestVersion: 'v1.3',
                parts: 'bot',
                unitTestsEnabled: true,
            })
            .withGenerators(testHelper.DEPENDENCIES);

})
```

@SBoudrias - any thoughts?

**Comment by Batta32** [comment]
Hi @wictorwilen,

We implemented a workaround for this issue in which we evaluated the version of **NodeJS** before running the tests.

```typescript
before( … ) {
   if(semver.gte(process.versions.node, `10.12.0`)) {
       run = false;
        …
     } else {
        …
        //GENERATOR LOGIC
        …
      }
}
describe(…) {
   if(!run) {
       this.skip();
    } 
   …
   // TEST LOGIC
   …
}
```

If you want to see the entirely code, check [here](https://github.com/microsoft/botframework-solutions/blob/5717335b0d91d17ef4b20be0ed2f4c3a45ca3d49/templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/test/generator-botbuilder-assistant.test.suite.js#L153).

**Comment by wictorwilen** [comment]
Thanks @Batta32 - it's a good workaround, but it would be great to be able to test on LTS versions of node. Node 8 support ends in only 6 months

**Comment by mshima** [comment]
Should be fixed with inquirer 7.0.2
